### PR TITLE
Bundle the perfetto build in assets/ along with the DevTools custom theme

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This PR:
- Bundles the Perfetto build output into our `assets/` directory. This allows us to view traces without an internet connection and allows us to apply a custom theme to the Perfetto UI. 
- Adds the ability to dynamically switch themes from light to dark in the Perfetto UI (files to review for this are under `assets/perfetto/dist/`)
- Adds a tool script that updates the bundled perfetto build from either the latest bits in the Perfetto repo or a custom build path

### CODE REVIEW

File: packages/devtools_app/assets/perfetto/dist/devtools_dark.css [hunk 1]
```diff
@@ -0,0 +1,164 @@
+/**
```
Comment (kenzieschmoll) [inline packages/devtools_app/assets/perfetto/dist/devtools_dark.css:1]: These are custom styles for the devtools dark theme

File: packages/devtools_app/assets/perfetto/dist/devtools_light.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/**
```
Comment (kenzieschmoll) [inline packages/devtools_app/assets/perfetto/dist/devtools_light.css:1]: these are custom styles for the devtools light theme

File: packages/devtools_app/assets/perfetto/dist/devtools_shared.css [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/**
```
Comment (kenzieschmoll) [inline packages/devtools_app/assets/perfetto/dist/devtools_shared.css:1]: these are custom styles for devtools, shared across light and dark themes

File: packages/devtools_app/assets/perfetto/dist/devtools_theme_handler.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
```
Comment (kenzieschmoll) [inline packages/devtools_app/assets/perfetto/dist/devtools_theme_handler.js:1]: JS to handle theme changes from DevTools and switch the CSS accordingly

File: packages/devtools_app/assets/perfetto/dist/index.html [hunk 5]
```diff
@@ -0,0 +1,121 @@
+<!doctype html>
+<html lang="en-us">
+<head>
+  <meta charset="utf-8">
+  <title>Perfetto UI</title>
+  <meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport" />
+  <link rel="shortcut icon" id="favicon" type="image/png" href="data:image/png;base64,iVBORw0KGgo=">
+  <link id="devtools-style" rel="stylesheet" href="devtools_dark.css">
+  <script src="devtools_theme_handler.js"></script>
```
Comment (kenzieschmoll) [inline packages/devtools_app/assets/perfetto/dist/index.html:9]: these two lines are the only custom additions to this index.html file. They are added by the `update_perfetto.sh` script

File: packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/_perfetto_controller_web.dart [hunk 6]
```diff
@@ -10,6 +10,7 @@ import 'dart:ui' as ui;
 
 import '../../../../../primitives/auto_dispose.dart';
 import '../../../../../primitives/trace_event.dart';
+import '../../../../../shared/globals.dart';
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/performance/panes/timeline_events/perfetto/_perfetto_controller_web.dart:13]: the remaining files should be reviewed

File: packages/devtools_app/assets/perfetto/dist/devtools_dark.css (hunk 1 above)
Comment (elliette) [inline packages/devtools_app/assets/perfetto/dist/devtools_dark.css:1]: [Optional] this and other CSS files - alphabetize property names (I use the "CSS Alphabetize" VSCode plugin for this)

File: packages/devtools_app/assets/perfetto/dist/devtools_theme_handler.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+  if (msgId === themePingId) {
+    const windowSource = messageEvent.source;
+    windowSource.postMessage('DART-DEVTOOLS-THEME-PONG', messageEvent.origin);
+    return;
+  }
+
+  const theme = messageEvent.data.theme;
+  if (theme !== 'light' && theme !== 'dark') {
+    console.warn("Cannot change Perfetto theme. Expected 'light' or 'dark' but got " + theme);
```
Comment (elliette) [inline packages/devtools_app/assets/perfetto/dist/devtools_theme_handler.js:26]: [Optional] Nicer string interpolation with backticks: \`Cannot change Perfetto theme. Expected 'light' or 'dark' but got $theme`

File: packages/devtools_app/assets/perfetto/dist/devtools_theme_handler.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    return;
+  }
+  var sheet = document.getElementById('devtools-style');
```
Comment (elliette) [inline packages/devtools_app/assets/perfetto/dist/devtools_theme_handler.js:29]: I think this can be `const`

**Review by elliette — APPROVED** [review]
Some optional suggestions but LGTM!

File: packages/devtools_app/assets/perfetto/dist/devtools_dark.css (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/assets/perfetto/dist/devtools_dark.css:1]: nice! thanks

**Comment by kenzieschmoll** [comment]
work towards https://github.com/flutter/devtools/issues/4207

**Pull request merged**